src/controllers/router-store/index.tsx

```diff
@@ -83,8 +91,14 @@ const actions: AllRouterActions = {
         routes,
         location: history.location,
         action: history.action,
+        loader,
       });
-      getResourceStore().actions.hydrate({ resourceContext, resourceData });
+
+      if (loader.hydrate) {
```

Comment:
It should still be possible to put this into the resource loader constructor time, and I don't think it's a big deal if you have to construct two separate loaders (one for the server, one for the client)